saleor/graphql/product/types/products.py

```diff
@@ -638,8 +644,15 @@ def resolve_channel_listing(root: ChannelContext[models.Product], info, **_kwarg
         return ProductChannelListingByProductIdLoader(info.context).load(root.node.id)
 
     @staticmethod
-    def resolve_collections(root: ChannelContext[models.Product], *_args):
-        return root.node.collections.all()
+    def resolve_collections(root: ChannelContext[models.Product], *_args, **_kwargs):
+        instances = root.node.collections.all()
+        channel_slug = root.channel_slug
+
+        collections = [
+            ChannelContext(node=collection, channel_slug=channel_slug)
+            for collection in instances
+        ]
+        return collections
```

Comment:
```suggestion
        return ChannelQsContext(qs=instances, channel_slug=channel_slug)
```

Reply:
After your changes i got error `AssertionError: User Error: expected iterable, but did not find one for field Product.collections.`. I will stay with my solution.